### ISSUE TITLE
Add a simple example notebook for state-to-state optimization

### ISSUE THREAD

**Issue by Basilewitsch** [qucontrol/krotov#19]
Add a trivial example notebook for the state-to-state optimization in a two-level system

**Comment by goerz** [comment]
Small nitpick: can we move the import on top of the introductory paragraph?

More importantly: I think this would be better with a shape function other than a constant 1. There is a flattop function defined in `krotov.shapes` that you could use to ensure the optimized control is zero at t=0 and t=T.

**Comment by goerz** [comment]
That's looking pretty good. I went over it to fix some minor typos etc, and to clarify two issues that I felt were still a bit confusing:

* I think people frequently don't really understand the role of S(t) as a shape *for the update*, not for the optimized control field. Of course, there's nothing fundamentally wrong about the guess field using S(t) for its initial shape as well, but this is not a requirement. I've rewritten the discussion in the notebook trying to be very explicit about that.
* The functional enters the optimization only indirectly, via the `chi_constructor`. The discussion was a bit misleading on this point and seemed to indicate that we needed to pass F to the optimization (via the info-hook). However, there is no requirement that the info-hook calculates or prints F, it's just convenient to track the progress of the optimization.

There is one last remaining (indirect) problem with the notebook: there seems to be something wrong with the propagation, because the norm of the state increases (which I noticed from the fidelity taking values greater than 1). For debugging, I'm now printing out the norm in the info-hook. We'll need an independent test (probably in `./tests/test_propagators`) to look into this. Could be worth a separate issue.
